pkg/trace/api/debug_server_serverless.go

```diff
@@ -0,0 +1,20 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build serverless
+// +build serverless
+
+package api
+
+import "github.com/DataDog/datadog-agent/pkg/trace/config"
+
+type DebugServer struct{}
+
+func NewDebugServer(conf *config.AgentConfig) *DebugServer {
```

Comment:
Consider logging a warning here letting customers know that the debug server is not enabled in a serverless environment.